### ISSUE TITLE
OnPropertyChanging not called for generated property

### ISSUE THREAD

**Issue by marhja** [CommunityToolkit/dotnet#711]
### Describe the bug

OnPropertyChanging is not called for a generated property when using the NotifyPropertyChangedFor attribute.

### Steps to reproduce

Using this class:

```csharp
public partial class MyObject : ObservableObject
{
  [ObservableProperty]
  [NotifyPropertyChangedFor(nameof(FullName))]
  private string name;

  [ObservableProperty]
  [NotifyPropertyChangedFor(nameof(FullName))]
  private string surname;

  public string FullName => $"{Name} {Surname}";

  protected override void OnPropertyChanging(PropertyChangingEventArgs e)
  {
    base.OnPropertyChanging(e);
    
    // This is never called for FullName
    Debug.Assert(e.PropertyName != nameof(FullName)); 
  }
}
```

The following code should cause OnPropertyChanging to be called for property FullName:

```csharp
var o = new MyObject();
o.Name = "Jack";
```

### Expected behavior

I expected `OnPropertyChanging` to be called also for property `FulleName` when Name is assigned a new value.

### IDE and version

VS 2022

### IDE version

17.2.12

### Nuget packages

- [ ] CommunityToolkit.Common
- [ ] CommunityToolkit.Diagnostics
- [ ] CommunityToolkit.HighPerformance
- [X] CommunityToolkit.Mvvm (aka MVVM Toolkit)

### Nuget package version(s)

8.2.0

### Help us help you

No, just wanted to report this

**Comment by Sergio0694** [comment]
Oooh good catch! Thank you for reporting this! 😄

**Comment by marhja** [comment]
OK, I can understand that this is by design, but then the following paragraph should probably be deleted from the doc comment for NotifyPropertyChangedForAttribute:

> If the containing type also implements the <see cref="INotifyPropertyChanging"/> interface and exposes a method with the same signature as <see cref="ObservableObject.OnPropertyChanging(string?)"/>, then this method will be invoked as well by the property setter.

**Comment by Sergio0694** [comment]
Thinking about this more, I think it makes sense to fix this. I've implemented it, but changed target to 9.0 instead of 8.2.1, as this is arguably a small breaking change, functionally speaking. Will want to validate this more with a preview release 🙂
